packages/synapse-react-client/src/mocks/MockSynapseContext.tsx

```diff
@@ -12,6 +14,12 @@ export const MOCK_CONTEXT_VALUE: SynapseContextType = {
   downloadCartPageUrl: '/DownloadCart',
   withErrorBoundary: false,
   keyFactory: new KeyFactory(MOCK_ACCESS_TOKEN),
+  synapseClient: new SynapseClient(
+    new Configuration({
+      fetchApi: fetchWithExponentialTimeout,
```

Comment:
Super fortunate that our existing abstraction just plugs in here. We pass in our `fetchWithExponentialTimeout` and we maintain the same behavior that we already have with automatic retries and all errors converted to our custom `SynapseClientError` class.